src/js/actions/libraries.js

```diff
@@ -1160,6 +1192,21 @@ define(function (require, exports) {
                 ELEMENT_COLORTHEME_TYPE
             ]
         });
+        
+        descriptor.addListener("placeEvent", function () {
```

Comment:
I think you need to assign both these handlers to private variables, so that they can be removed upon reset.  See other handlers in on other actions as an example.